client/client.go

```diff
@@ -1216,12 +1208,14 @@ func (c *Client) syncConsul() {
 					if taskState.State == structs.TaskStateRunning {
 						if tr, ok := ar.tasks[taskName]; ok {
 							for _, service := range tr.task.Services {
-								services[service.ID(ar.alloc.ID, tr.task.Name)] = struct{}{}
+								svcIdentifier := fmt.Sprintf("%s-%s", ar.alloc.ID, tr.task.Name)
```

Comment:
@diptanu Would this be a problem for inplace upgrades of Nomad if we change the scheme?